sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -179,23 +179,34 @@ CosmosAsyncContainer setId(String id) {
      * successful completion will contain a single resource response with the
      * created cosmos item. In case of failure the {@link Mono} will error.
      *
+     * @param <T> the type parameter
      * @param item the cosmos item represented as a POJO or cosmos item object.
+     * @param partitionKey the partition key
      * @param options the request options.
-     * @return an {@link Mono} containing the single resource response with the
-     * created cosmos item or an error.
+     * @return an {@link Mono} containing the single resource response with the created cosmos item or an error.
      */
-    public Mono<CosmosAsyncItemResponse> createItem(Object item, CosmosItemRequestOptions options) {
+    public <T> Mono<CosmosAsyncItemResponse<T>> createItem(T item,
+                                                           PartitionKey partitionKey,
+                                                           CosmosItemRequestOptions options) {
+        if (options == null) {
+            options = new CosmosItemRequestOptions();
+        }
+        options.setPartitionKey(partitionKey);
```

Comment:
What will be the behavior if `partitionKey` is passed as null ?
I see a couple of options here. 
Either default it to `PartitionKey.NONE` or use something like this `checkNoNull(partitionKey, "Partition Key should not be null")` 

Please make sure this behavior is consistent across all the APIs.

Reply:
null is a valid partitionkey. Also, please note that null partitionkey and PartitionKey.NONE are totally different